tests/static_analyzer/test_csharp_adapter.py

```
@@ -365,11 +417,15 @@ def fake_run(*_args, **_kwargs):
             "static_analyzer.engine.adapters.csharp_adapter.subprocess.run",
             fake_run,
         )
-        CSharpAdapter().prepare_project(tmp_path)
+        monkeypatch.setattr(
+            "static_analyzer.engine.adapters.csharp_adapter.resolve_dotnet_sdk",
+            lambda _root: (_ for _ in ()).throw(DotnetSdkError("compatible .NET SDK was not found")),
+        )
+        with pytest.raises(RuntimeError, match="compatible .NET SDK"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)